test/async_scope_v1_test.cpp

```diff
@@ -0,0 +1,661 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/allocate.hpp>
+#include <unifex/any_sender_of.hpp>
+#include <unifex/v1/async_scope.hpp>
+#include <unifex/just.hpp>
+#include <unifex/just_done.hpp>
+#include <unifex/just_from.hpp>
+#include <unifex/just_void_or_done.hpp>
+#include <unifex/let_value_with.hpp>
+#include <unifex/let_value_with_stop_source.hpp>
+#include <unifex/let_value_with_stop_token.hpp>
+#include <unifex/never.hpp>
+#include <unifex/on.hpp>
+#include <unifex/scope_guard.hpp>
+#include <unifex/sequence.hpp>
+#include <unifex/single_thread_context.hpp>
+#include <unifex/sync_wait.hpp>
+#include <unifex/then.hpp>
+#include <unifex/when_all.hpp>
+
+#include "mock_receiver.hpp"
+#include "stoppable_receiver.hpp"
+
+#include <gtest/gtest.h>
+
+#include <array>
+#include <atomic>
+#include <optional>
+
+using namespace unifex;
+using namespace unifex_test;
+
+namespace {
+
+struct signal_on_destruction {
+  async_manual_reset_event* destroyed_;
+
+  signal_on_destruction(async_manual_reset_event* destroyed) noexcept
+    : destroyed_(destroyed) {}
+
+  signal_on_destruction(signal_on_destruction&& other) noexcept
+    : destroyed_(std::exchange(other.destroyed_, nullptr)) {}
+
+  ~signal_on_destruction() {
+    if (destroyed_) {
+      destroyed_->set();
+    }
+  }
+};
+
+}  // namespace
+
+struct async_scope_test : testing::Test {
+  async_scope scope;
```

Comment:
Perhaps this should be explicit about the version.
```suggestion
  v1::async_scope scope;
```